### PR TITLE
I18n: Fix Gutenberg strings appearing in English for pt-BR locale

### LINKED ISSUE #18161: I18n: Some textual content remains in English when using WordPress app in PT-BR language
### Expected behavior
While using the WordPress app in iOS devices configured with PT-BR language, textual content should be translated accordingly.

### Actual behavior
Some textual content remains in English when using WordPress app in PT-BR language (iOS device, iPhone XS).
> I just explored the content of `Blog Post` creation. There might be much more untranslated content. Please, let me know if you'll need me to run another session to explore this even more and provide a complete report.

| Adding images | Text options | Paragraph placeholder | Paragraph settings |
| --- | --- | --- | --- |
|Adding images|Text options|Paragraph placeholder|Paragraph settings|

| `Block removed` toast | Color labels | Font sizing options | General placeholders |
| --- | --- | --- | --- |
|Block removed toast|IColor labels|Font sizing options|General placeholders

### Steps to reproduce the behavior

1. Switch the device's default language to PT-BR
2. Navigate throughout the app exploring proper translation

### PR DESCRIPTION
**Pull request by staskus**
Fixes #18161

## Description

Some strings in the editor appear in English instead of Brazilian Portuguese. It happens because instead of using `pt-BR` localization file we pass  `pt` location file which contains much fewer translations.

`currentLProjFolderName` method in `GutenbergViewController+Localization` manually removes "region" part of the identifier which makes Gutenberg use different translations than the app uses.

### Solution

Use `bundle.preferredLocalizations` which takes into account:
- User preferences
- Localizations contained in the bundle

and returns them sorted in order. In practice it matches the behaviour of `NSLocalizedString`. 

For example, for `en_IN` locale previous solution would use `en` strings which are in American English.
However `NSLocalizedStrings` and `bundle.preferredLocalizations` use `en-gb` localizations. 

### Considered solutions

I planned to expand the logic of the current approach by first checking if there's Localization.strings for a full `Locale.identifier`  before returning `identifierWithoutRegion`. However, `Localization.strings` path uses dashes for identifiers (e.g.`pt-BR`) while `Locale.identifier` uses underscores (e.g. `pt_BR`), which would require additional identifier logic. Also with such a solution, we would make assumptions that a specific region always defaults to a locale without a region which is not necessarily true. It's safer to use existing Apple-developed logic.

## Test

1. Select Português (Brasil) language with a Brazil region
2. Open WordPress app and log in
3. Create a new post ("Post de blog")
4. Notice "Comece a escrever" written in the editor (instead of "Start writing" as before)

## Regression Notes
1. Potential unintended areas of impact

Gutenberg localization for other languages and regions.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manually testing with multiple supported and non-supported languages and regions.

3. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

image

### CODE REVIEW

**Comment by staskus** [comment]
@fluiddot I saw you being active in the original issue so I'm assigning you to check the fix as well. Thank you!

**Review by sla8c — APPROVED** [review]
This LGTM.  I think it would still be worth @fluiddot reviewing as a second pair of eyes before merging since I'm not familiar with WP app's localization.

**Comment by staskus** [comment]
> This LGTM. I think it would still be worth @fluiddot reviewing as a second pair of eyes before merging since I'm not familiar with WP app's localization.

Thanks for the review! I agree that it's worth for a second pair of eyes to take a look at it as well 👍

**Comment by fluiddot** [comment]
Thank you very much @staskus for fixing this 🎊 !

> > This LGTM. I think it would still be worth @fluiddot reviewing as a second pair of eyes before merging since I'm not familiar with WP app's localization.
> 
> Thanks for the review! I agree that it's worth for a second pair of eyes to take a look at it as well 👍

Yeah, I'm more than happy to check this but I'm afraid I won't be able to take a look until tomorrow (Wednesday). If that's ok, I'll share my review tomorrow as soon as possible.

**Comment by staskus** [comment]
> Thank you very much @staskus for fixing this 🎊 !
> 
> 
> 
> > > This LGTM. I think it would still be worth @fluiddot reviewing as a second pair of eyes before merging since I'm not familiar with WP app's localization.
> 
> > 
> 
> > Thanks for the review! I agree that it's worth for a second pair of eyes to take a look at it as well 👍
> 
> 
> 
> Yeah, I'm more than happy to check this but I'm afraid I won't be able to take a look until tomorrow (Wednesday). If that's ok, I'll share my review tomorrow as soon as possible.

Thank you. There's no rush, I'll wait for the review!

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

I've tested several languages/regions within the editor and works as expected, I confirm that using Portuguese(Brazil) displays all the translations. I'd like to note that using Portuguese(Portugal) still shows some strings untranslated but as mentioned [here](https://github.com/wordpress-mobile/WordPress-iOS/issues/18161#issuecomment-1074997093), this is expected because [WordPress for iOS GlotPress project](https://translate.wordpress.org/projects/apps/ios/dev/) has a low translation coverage for that locale.

Thank you very much @staskus for fixing this issue 🙇 !

**Review by guarani — APPROVED** [review]
👋 @fluiddot, since 20.8 is being cut next Monday, the issue being fixed is labeled high priority, and I understand that @staskus is away till Monday, would it make sense if I went ahead and merged this?

I'm not sure a review from me is needed at this stage since @sla8c and @fluiddot have already reviewed, so I reviewed just to be on the safe side. 

I agree `bundle.preferredLocalizations.first` is a good choice. In my testing, it returns an subset of the app's supported localizations, ordered from most preferable (i.e. closest match to the device's current language) to least preferable. Taking Portuguese (Brazilian) for example, it returns `["pt-BR", "pt"]`. If I switch to Portuguese (Portugal) it returns `["pt"]`, which I think makes sense. The hyphen here matches the hyphen used in the localized strings filename: https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Resources/pt-BR.lproj/Localizable.strings.

**Comment by fluiddot** [comment]
> 👋 @fluiddot, since 20.8 is being cut next Monday, the issue being fixed is labeled high priority, and I understand that @staskus is away till Monday, would it make sense if I went ahead and merged this?

Sure thing, feel free to go ahead and merge this, thanks 🙇 !

**Comment by fluiddot** [comment]
> 👋 @fluiddot, since 20.8 is being cut next Monday, the issue being fixed is labeled high priority, and I understand that @staskus is away till Monday, would it make sense if I went ahead and merged this?

@guarani Sure thing, feel free to go ahead and merge this, thanks 🙇 !

**Comment by staskus** [comment]
Thank you for the reviews! I'm merging it 🚀

**Pull request merged**
